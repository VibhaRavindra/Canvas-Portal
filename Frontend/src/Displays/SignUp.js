import React, { Component } from 'react';
import '../App.css';
import { Redirect } from 'react-router-dom';

class SignUp extends Component {

    constructor(props){
        super(props);
        this.state = {
            isSignedUp: false,
            isInvalid: "",
            errMessage: ""
        }
        this.renderSignIn = this.renderSignIn.bind(this);
        this.submitSignUp = this.submitSignUp.bind(this);
    }
    renderSignIn(){
        console.log("renderSignIn")
        this.setState({
            isSignedUp: true
        })
    }
    async submitSignUp(e) {
        e.preventDefault();
        const data = new FormData(e.target);
        console.log(data.values);
        const signUpUser = await fetch('/addUser',{
            body: data,
            method: 'PUT'
        })
        const responseSignUp = await signUpUser.json();
        this.setState({
            isInvalid: responseSignUp.isInvalid,
            errMessage: responseSignUp.errMessage
        })
        console.log(responseSignUp);
    }
    async componentDidMount(){
        const signInUser = await fetch('/isLoggedIn',{
            method: 'GET'
        })
        const responseSignIn = await signInUser.json();
        this.setState({
            isSignedUp: responseSignIn.isLoggedIn
        })
    }
    render(){
        let renderSignIn, failCase, renderDashboard;
        if(this.state.isSignedUp){
            renderSignIn = <Redirect to='/signin' />
        }else if(this.state.isInvalid === true){
            failCase = <div className="error-box">
            <span>
            <img className="error-icon" src="https://www.freeiconspng.com/uploads/alert-icon-red-11.png" alt="" />
            </span>
            <span className="error-msg">{this.state.errMessage}</span>
        </div>
        }else if (this.state.isInvalid === false){
            renderDashboard = <Redirect to='/signin' />
        }
        return(
            <div>
            {renderDashboard}
            <div className="LoginPage">
                <div className="imageHeader">
                    <img className="sjsuImage" src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7" alt="" />
                    <div className="beacon-container"></div>
                </div>
                
                <div className="LogIncontainer">
                    <p className="signInText">Sign Up</p>
                    {failCase}
                    <form className="form"  onSubmit={this.submitSignUp}>
                    <input type="text" className="inputStyle1" name="username" placeholder="Name" required />  
                    <input type="email" className="inputStyle1" name="emailId" placeholder="SJSU Email ID" required />                
                    <input type="password" className="inputStyle1" name="password" placeholder="Password" required /> <br />
                    Student<input type="radio" value="student" name="employee" checked/>  
                    Faculty<input type="radio" value="faculty" name="employee" /> 
                    <button type="submit" className="signInBtn">Sign Up</button>
                    </form>
                    <button className="orText"  onClick={this.renderSignIn}>Already signed up?</button>
                    {renderSignIn}
                </div> 
            </div>
            </div>
        )
    }
}

export default SignUp;