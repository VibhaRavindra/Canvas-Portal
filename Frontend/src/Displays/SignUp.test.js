import React from 'react';
import SignUp from './SignUp';
import renderer from 'react-test-renderer';
import { mount, shallow, render } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('Signup renders correctly', () => {
    const tree = renderer
      .create(<SignUp/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });