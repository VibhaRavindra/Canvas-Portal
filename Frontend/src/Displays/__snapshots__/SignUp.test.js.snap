// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in renders correctly 1`] = `
<div>
  <div
    className="LoginPage"
  >
    <div
      className="imageHeader"
    >
      <img
        alt=""
        className="sjsuImage"
        src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7"
      />
      <div
        className="beacon-container"
      />
    </div>
    <div
      className="LogIncontainer"
    >
      <p
        className="signInText"
      >
        Sign In
      </p>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="inputStyle1"
          name="emailId"
          placeholder="SJSU Email ID"
          required={true}
          type="email"
        />
        <input
          className="inputStyle1"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
        />
        <button
          className="signInBtn"
          type="submit"
        >
          Sign In
        </button>
        <button
          className="orText"
          onClick={[Function]}
        >
          Create an account? 
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Signup renders correctly 1`] = `
<div>
  <div
    className="LoginPage"
  >
    <div
      className="imageHeader"
    >
      <img
        alt=""
        className="sjsuImage"
        src="https://ok2static.oktacdn.com/bc/image/fileStoreRecord?id=fs01heub3azJBMXWF0x7"
      />
      <div
        className="beacon-container"
      />
    </div>
    <div
      className="LogIncontainer"
    >
      <p
        className="signInText"
      >
        Sign Up
      </p>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="inputStyle1"
          name="username"
          placeholder="Name"
          required={true}
          type="text"
        />
        <input
          className="inputStyle1"
          name="emailId"
          placeholder="SJSU Email ID"
          required={true}
          type="email"
        />
        <input
          className="inputStyle1"
          name="password"
          placeholder="Password"
          required={true}
          type="password"
        />
         
        <br />
        Student
        <input
          checked={true}
          name="employee"
          type="radio"
          value="student"
        />
        Faculty
        <input
          name="employee"
          type="radio"
          value="faculty"
        />
        <button
          className="signInBtn"
          type="submit"
        >
          Sign Up
        </button>
      </form>
      <button
        className="orText"
        onClick={[Function]}
      >
        Already signed up?
      </button>
    </div>
  </div>
</div>
`;
